arwen/host/asyncCall.go

```diff
@@ -390,20 +407,37 @@ func (host *vmHost) createCallbackContractCallInput(
 	}
 
 	if isESDTOnCallBack {
```

Comment:
Suggest splitting this code block into small functions.

Reply:
simplified this.